kork-pubsub-aws/src/test/java/com/netflix/spinnaker/kork/pubsub/aws/SQSSubscriberTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2024 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.kork.pubsub.aws;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.model.SubscribeResult;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.GetQueueUrlResult;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.netflix.spectator.api.DefaultRegistry;
+import com.netflix.spinnaker.kork.pubsub.aws.api.AmazonMessageAcknowledger;
+import com.netflix.spinnaker.kork.pubsub.aws.api.AmazonPubsubMessageHandler;
+import com.netflix.spinnaker.kork.pubsub.aws.config.AmazonPubsubProperties;
+import java.util.List;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class SQSSubscriberTest {
+
+  SQSSubscriber subscriber;
+  AmazonMessageAcknowledger messageAcknowledger;
+
+  @Test
+  @DisplayName("acknowledger acknowledges when handler succeeds")
+  void testAcknowledgerAcksHandlerSucceeds() {
+    // given
+    messageAcknowledger = mock(AmazonMessageAcknowledger.class);
+    subscriber =
+        new SQSSubscriber(
+            subscription(),
+            mock(AmazonPubsubMessageHandler.class),
+            messageAcknowledger,
+            amazonSNS(),
+            amazonSQS(),
+            enableOnce(),
+            new DefaultRegistry());
+
+    // when
+    subscriber.initializeQueue();
+    subscriber.listenForMessages();
+
+    // then
+    verify(messageAcknowledger, atLeastOnce()).ack(any(), any());
```

Comment:
Doesn't the groovy test do
```suggestion
    verify(messageAcknowledger, once()).ack(any(), any());
```

Reply:
Updated the code.